store/src/java/com/zimbra/cs/lmtpserver/ExternalEmailWarning.java

```diff
@@ -173,26 +185,38 @@ public String getUpdatedContent(String content) {
         if (content != null) {
             final Matcher ctTextPlainMatcher = CONTENT_TYPE_TEXT_PLAIN_PATTERN.matcher(content);
             if (ctTextPlainMatcher.find()) {
+                String textPlainHeader = ctTextPlainMatcher.group();
                 final int end = ctTextPlainMatcher.end();
-                final StringBuilder sb = new StringBuilder();
-                sb.append(content.substring(0, end));
-                if (getTextPlainWarning() != null)
-                    sb.append(getTextPlainWarning());
-                sb.append(content.substring(end));
-                content = sb.toString();
+                if (textPlainHeader.contains("Content-Transfer-Encoding: base64")) {
+                    content = appendWarning(content, end, getEncodedTextPlainWarning());
+                } else {
+                    content = appendWarning(content, end, getTextPlainWarning());
+                }
             }
             final Matcher ctTextHtmlMatcher = CONTENT_TYPE_TEXT_HTML_PATTERN.matcher(content);
+            final Matcher ctTextHtmlWithoutBodyTagMatcher = CONTENT_TYPE_TEXT_HTML_WIHOUT_BODY_TAG_PATTERN.matcher(content);
             if (ctTextHtmlMatcher.find()) {
                 final int end = ctTextHtmlMatcher.end();
-                final StringBuilder sb = new StringBuilder();
-                sb.append(content.substring(0, end));
-                if (getTextHtmlWarning() != null)
-                    sb.append(getTextHtmlWarning());
-                sb.append(content.substring(end));
-                content = sb.toString();
+                content = appendWarning(content, end, getTextHtmlWarning());
+            } else if (ctTextHtmlWithoutBodyTagMatcher.find()) {
+                String textHtmlHeader = ctTextHtmlWithoutBodyTagMatcher.group();
+                final int end = ctTextHtmlWithoutBodyTagMatcher.end();
+                if (textHtmlHeader.contains("Content-Transfer-Encoding: base64")) {
```

Comment:
add null check for textHtmlHeader to avoid NullPoinerException

Reply:
Added null check for textHtmlHeader